cmd/install/assets/hypershift_operator.go

```diff
@@ -725,3 +725,146 @@ func (r HypershiftRecordingRule) Build() *prometheusoperatorv1.PrometheusRule {
 	rule.Spec = recordingRuleSpec()
 	return rule
 }
+
+type HyperShiftClientClusterRole struct{}
+
+func (o HyperShiftClientClusterRole) Build() *rbacv1.ClusterRole {
+	role := &rbacv1.ClusterRole{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "ClusterRole",
+			APIVersion: rbacv1.SchemeGroupVersion.String(),
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "hypershift-client",
+		},
+		Rules: []rbacv1.PolicyRule{
+			{
+				APIGroups: []string{"hypershift.openshift.io"},
+				Resources: []string{"hostedclusters", "nodepools"},
+				Verbs:     []string{"*"},
+			},
+		},
+	}
+	return role
+}
+
+type HyperShiftClientServiceAccount struct {
+	Namespace *corev1.Namespace
+}
+
+func (o HyperShiftClientServiceAccount) Build() *corev1.ServiceAccount {
+	sa := &corev1.ServiceAccount{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "ServiceAccount",
+			APIVersion: corev1.SchemeGroupVersion.String(),
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: o.Namespace.Name,
+			Name:      "hypershift-client",
+		},
+	}
+	return sa
+}
+
+type HyperShiftClientClusterRoleBinding struct {
+	ClusterRole    *rbacv1.ClusterRole
+	ServiceAccount *corev1.ServiceAccount
+	GroupName      string
+}
+
+func (o HyperShiftClientClusterRoleBinding) Build() *rbacv1.ClusterRoleBinding {
+	binding := &rbacv1.ClusterRoleBinding{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "ClusterRoleBinding",
+			APIVersion: rbacv1.SchemeGroupVersion.String(),
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "hypershift-client",
+		},
+		RoleRef: rbacv1.RoleRef{
+			APIGroup: "rbac.authorization.k8s.io",
+			Kind:     "ClusterRole",
+			Name:     o.ClusterRole.Name,
+		},
+		Subjects: []rbacv1.Subject{
+			{
+				Kind:      "ServiceAccount",
+				Name:      o.ServiceAccount.Name,
+				Namespace: o.ServiceAccount.Namespace,
+			},
+			{
+				Kind:     "Group",
+				APIGroup: "rbac.authorization.k8s.io",
+				Name:     o.GroupName,
+			},
+		},
+	}
+	return binding
+}
+
+type HyperShiftReaderClusterRole struct{}
+
+func (o HyperShiftReaderClusterRole) Build() *rbacv1.ClusterRole {
+	clusterOperatorRole := HyperShiftOperatorClusterRole{}.Build()
```

Comment:
If we change HyperShiftOperatorClusterRole we won't necessarily know that this is impacted. My preference would be to explicitly include the RBAC rules here, even if it means copy-pasting code.

Reply:
addressed in 99eb9c8